### DISCUSSION TITLE
[Help wanted] - Renovate configurations to override/replace the computed newVersion

### DISCUSSION THREAD

**Discussion by mvgadagi** [renovatebot/renovate#31578] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

38.70.3

### Please tell us more about your question or problem

I’m currently using a self-hosted Renovate setup to manage and update dependencies in my repositories. Recently, I’ve encountered a new scenario and I would appreciate any guidance or possible solutions to solve the issues

Current Setup
In my repository, I have dependencies listed in dependency.xml with the following format:

```
<dependency name="a" version="1-0-0-0"/>
<dependency name="b" version="1-0-1-0"/>
```

Available Versions
The new versions of the dependencies are available in our Artifactory, and their versioning is as follows:
```
Dependency a
1.0.0.0
1.0.0.1

Dependency b
1.0.1.0
1.0.1.1
```

The Issue
When Renovate updates the dependencies, it computes the new versions correctly in terms of major, minor, and patch versions. However, the version format in my dependency.xml is slightly different (hyphen-separated), and I would like the updated versions to maintain that format.

Current Behavior:

```
<dependency name="a" version="1.0.0.1"/>
<dependency name="b" version="1.0.1.1"/>
```

Expected Behavior:
```
<dependency name="a" version="1-0-0-1"/>
<dependency name="b" version="1-0-1-1"/>
```

Current Renovate configurations

```{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "regexManagers": [
    {
      "fileMatch": [
        "(?i)^.*(dependency.xml)$"
      ],
      "matchStrings": [
        "(<(dependency).*)(name=\"(?<depName>.*?)\").*(version=\"(?<currentValue>.*?)\")(.*/>)"
      ],
      "versioningTemplate": "regex:^(?<major>\\d+)(-|\\.)(?<minor>\\d+)(-|\\.)(?<patch>\\d+)(-|\\.)(?<build>\\d+)$",
      "datasourceTemplate": "artifactory"
    }
  ],
  "packageRules": [
    {
      "description": "dependencies",
      "matchDatasources": [
        "artifactory"
      ],
      "matchPaths": [
        "dependency.xml"
      ],
      "registryUrls": [
        "https://myregistry.com/dependencies"
      ]
    }
  ]
}
```

Is there a way to configure Renovate so that the new versions are compatible with the hyphenated format in my dependency.xml rather than using the version format directly from the data source?

Any help or suggestions would be greatly appreciated!

Thanks a lot in advance !!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
I'm not aware of any way to achieve this unless the idea described in https://github.com/renovatebot/renovate/discussions/31372 is implemented
